numcodecs/tests/test_zarr3.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import annotations
+from typing import Iterator
+
+
+import numpy as np
+import pytest
+import sys
+
+from numcodecs.registry import get_codec
+
+try:
+    from zarr.codecs.registry import get_codec_class
+    from zarr.array import Array
+    from zarr.common import JSON
+    from zarr.codecs import BytesCodec
+    from zarr.abc.store import Store
+    from zarr.store import MemoryStore, StorePath
+
+except ImportError:
+    pass
+
+
+pytestmark = pytest.mark.skipif(
+    sys.version_info < (3, 10), reason="zarr-python 3 requires Python 3.10 or higher"
+)
+
+
+@pytest.fixture
+def store() -> Iterator[Store]:
+    yield StorePath(MemoryStore(mode="w"))
+
+
+@pytest.mark.parametrize(
+    "codec_id", ["blosc", "lz4", "zstd", "zlib", "gzip", "bz2", "lzma", "shuffle"]
+)
+def test_generic_codec(store: Store, codec_id: str):
+    data = np.arange(0, 256, dtype="uint16").reshape((16, 16))
+
+    with pytest.warns(UserWarning, match="Numcodecs.*"):
+        a = Array.create(
+            store / "generic",
+            shape=data.shape,
+            chunk_shape=(16, 16),
+            dtype=data.dtype,
+            fill_value=0,
+            codecs=[
+                BytesCodec(),
+                get_codec_class(f"numcodecs.{codec_id}")({"id": codec_id}),
+            ],
+        )
+
+    a[:, :] = data.copy()
+    assert np.array_equal(data, a[:, :])
```

Comment:
```suggestion
   np.testing.assert_array_equal(data, a[:, :])
```
I think will give more information if the test fails?